rust/src/core/ffi.rs

```diff
@@ -701,6 +713,48 @@ pub extern "C" fn opendp_core__new_function(
     FfiResult::Ok(util::into_raw(Function::new_fallible(wrap_func(function))))
 }
 
+#[bootstrap(
+    features("contrib"),
+    arguments(function(rust_type = "$pass_through(TO)")),
+    dependencies("c_function")
+)]
+/// Construct a Function from a user-defined callback.
+/// This is meant for internal use, as it does not require "honest-but-curious",
+/// unlike `new_function`.
+///
+/// Can be used to build a post-processor.
+///
+/// `function` must meet two criteria.
+///
+/// First, `function` must not use global state.
+/// For instance, a postprocessor that accesses the system clock time
+/// can be used to build a measurement that reveals elapsed execution time,
+/// which escalates a side-channel vulnerability into a direct vulnerability.
+///
+/// Secondly, `function` must only raise data-independent exceptions.
+/// For instance, raising an exception with the value of a DP release will both
+/// reveal the DP output and cancel the computation, potentially avoiding privacy accounting.
+///
+/// # Arguments
+/// * `function` - A function mapping data to a value of type `TO`
+///
+/// # Generics
+/// * `TO` - Output Type
+#[allow(dead_code)]
+fn _new_pure_function<TO>(function: CallbackFn) -> Fallible<AnyFunction> {
+    let _ = function;
+    panic!("this signature only exists for code generation")
+}
```

Comment:
Is there anything that needs this now? Or is it just in place so that when we do need it, it's available? It feels like a backdoor.

Reply:
Moved into the internal module. Nothing uses it now.